packages/dds/tree/src/test/events/events.spec.ts

```diff
@@ -124,38 +147,43 @@ describe("EventEmitter", () => {
 		assert.strictEqual(count, 0);
 	});
 
-	// Note: This behavior is not contractually required (see docs for `Listenable.on()`),
-	// but is tested here to check for changes or regressions.
-	it("correctly handles multiple registrations of the same listener", () => {
+	it("errors on multiple registrations of the same listener", () => {
 		const emitter = createEmitter<TestEvents>();
-		let count: number;
+		let count = 0;
 		const listener = () => (count += 1);
-		const off1 = emitter.on("open", listener);
-		const off2 = emitter.on("open", listener);
-
-		count = 0;
-		emitter.emit("open");
-		assert.strictEqual(count, 2); // Listener should be fired twice
-
-		count = 0;
-		off1();
+		emitter.on("open", listener);
+		assert.throws(
+			() => emitter.on("open", listener),
+			(e: Error) => validateAssertionError(e, /register.*twice.*open/),
+		);
+		// If error is caught, the listener should still fire once for the first registration
 		emitter.emit("open");
 		assert.strictEqual(count, 1);
+	});
 
-		count = 0;
-		off2();
-		emitter.emit("open");
-		assert.strictEqual(count, 0);
+	it("allows registrations of event names that are symbols", () => {
```

Comment:
should the multi-registration be broken into separate case or add comments about the different aspects tested with comments in the case

Reply:
I added a comment enumerating the handful of things covered by this test. I don't think it's worth making another test case for double-registration of symbols, since we already have test coverage of double-registration in general, but I did want to check that the error message correctly prints out the name of the symbol.